### PR TITLE
chore: rework release note handling

### PR DESCRIPTION
**Pull request by jakobmoellerdev**
#### What this PR does / why we need it

reworks release note management based on github release note api

It does this in the following way:
1. Creates a tag on cutoff that is annotated
2. Calls the Github Release Note API together with a release note configuration file (very similar to drafter but allows us to specify a previous tag)
3. Generates the notes for the current branch based on the last found tag.

This has the advantage that we can explicitly determine which release notes are generated from when to when and also use the contributors fields.

TODO:
1. Test on other repo
2. Make sure that the Merge commit in main that we used for testing disappears.

### CODE REVIEW

File: .github/workflows/release-drafter.yaml [hunk 1]
```diff
@@ -31,12 +31,69 @@ jobs:
...
\ No newline at end of file
+          fetch-depth: 0
+          fetch-tags: true
+      - name: Setup Release with gh
+        env:
+          REF: ${{ github.ref }}
+          REPO: ${{ github.repository }}
+          GH_TOKEN: ${{ steps.generate_token.outputs.token }}
+        run: |
+          git fetch --tags
```
Comment (hilmarf) [inline .github/workflows/release-drafter.yaml:51]: Does the actions/checkout@v4 `fetch-tags: true` not work? Or why is there another `git fetch --tags` necessary?

File: .github/workflows/release-drafter.yaml [hunk 2]
```diff
@@ -31,12 +31,69 @@ jobs:
...
+          notes=$(\
+             gh api  \
+               --method POST \
+               -H "Accept: application/vnd.github+json" \
+               -H "X-GitHub-Api-Version: 2022-11-28" \
+               /repos/${REPO}/releases/generate-notes \
+               -f "tag_name=${{env.RELEASE_VERSION}}" \
+               -f "target_commitish=${{ github.ref }}" \
+               -f "previous_tag_name=${previous_tag}" \
+               -f "configuration_file_path=.github/release.yml" \
```
Comment (hilmarf) [inline .github/workflows/release-drafter.yaml:77]: When we do have the option to specify which config to use... can we then maybe place the config under `.github/config`?

**Review by hilmarf — DISMISSED** [review]

File: .github/workflows/release-drafter.yaml (hunk 1 above)
Comment (jakobmoellerdev) [reply .github/workflows/release-drafter.yaml:51]: Leftover from testing, not necessary

File: .github/workflows/release-drafter.yaml (hunk 2 above)
Comment (jakobmoellerdev) [reply .github/workflows/release-drafter.yaml:77]: My original thought was that I would like to keep the default location as marked in the official documentation but im also fine to adopt it. WDYT?

Comment (hilmarf) [reply .github/workflows/release-drafter.yaml:77]: @frewilhelm ?

Comment (jakobmoellerdev) [reply .github/workflows/release-drafter.yaml]: adjusted. changed my mind :D

**Review by hilmarf — DISMISSED** [review]

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -181,8 +181,22 @@ jobs:
         git commit -m "Release $RELEASE_VERSION"
         msg="Release ${{ env.RELEASE_VERSION }}"
         git tag --annotate --message "${msg}" ${{ env.RELEASE_VERSION }}
-        # push both the tag as well as a release branch with that tag.
+        # push the tag.
         git push origin ${{ env.RELEASE_VERSION }}
+    # in case we do not have a release candidate, we should merge the release into the branch
+    # this is so that the tag shows up in the branch and we can use it for the release notes.
```
Comment (frewilhelm) [inline .github/workflows/release.yaml:187]: This sound more complicated that is should be. What do we merge in which branch?

**Review by hilmarf — APPROVED** [review]

**Review by frewilhelm — APPROVED** [review]

**Pull request merged**